crates/ruma-state-res/src/event_auth.rs

```diff
@@ -561,10 +561,19 @@ where
     E: Event,
     F: Fn(&EventType, &str) -> Option<Arc<E>>,
 {
-    let power_event_state_key = power_event.state_key().expect("power events have state keys");
-    let current_state = if let Some(current_state) =
-        fetch_state(&power_event.event_type(), &power_event_state_key)
-    {
+    match power_event.state_key().as_deref() {
+        Some("") => {}
+        Some(key) => {
+            error!("m.room.power_levels event has non-empty state key: {}", key);
+            return None;
+        }
+        None => {
+            error!("check_power_levels requires an m.room.power_levels *state* event argument");
+            return None;
+        }
+    }
+
+    let current_state = if let Some(current_state) = fetch_state(&power_event.event_type(), "") {
```

Comment:
Looks like ypu don't even need the power event passed into this function. We should remove that parameter at some point